drizzlepac/runastrodriz.py

```diff
@@ -305,10 +305,14 @@ def process(inFile, force=False, newpath=None, inmemory=False, num_cores=None,
             _cal_prodname = inFilename
             _new_asn.extend(_inlist)  # kept so we can delete it when finished
 
-        # check to see whether FLC files are also present, and need to be updated
-        # generate list of FLC files
+            # check to see whether FLC files are also present, and need to be updated
+            # generate list of FLC files
+            _calfiles_flc = [f.replace('_flt.fits', '_flc.fits')
+                             for f in _calfiles
+                             if os.path.exists(f.replace('_flt.fits', '_flc.fits'))]
```

Comment:
For improved efficiency it is better to avoid replacing `flt`->`flc` _twice_. I think this could be done in the following way:
```python
_calfiles_flc = [x for x in [f.replace('_flt.fits', '_flc.fits') for f in _calfiles] if os.path.exists(x)]
```

Reply:
As we discussed, the nested list comprehension calls is elegant, but not necessarily any clearer or easier to maintain.  So, unless there is a compelling reason, the current implementation will remain for the sake of clarity in the code.